rest/server_context.go

```diff
@@ -408,6 +408,12 @@ func (sc *ServerContext) _getOrAddDatabaseFromConfig(config DatabaseConfig, useE
 		return nil, err
 	}
 
+	if spec.Scope != nil && spec.Collection != nil {
```

Comment:
`Scope` and `Collection` on a `BucketSpec` are temporary and go away as soon as we need to support more than one collection.

We should be checking `len(DatabaseConfig.Scopes) > 0` to determine whether we're running with any named scope/collection, even if they happen to be `_default`.`_default`

Reply:
Will do - the only downside is that that means this check will happen after DbContext setup and index build, which means that you'll be waiting a while before finding out that your configuration is unsupported. Not a huge deal though.

Reply:
I dunno how we're able to build indexes on a collection if the server doesn't support collections

`config DatabaseConfig` in this function contains the nessesary info to determine whether the database is going ot be using a named collection, without creating a DatabaseContext first

Reply:
The index build would fail, but it'd be good if we gave a descriptive error than some N1QL internals. This is all moot though, because I misread `DatabaseConfig` as `DatabaseContext`...